apps/wallet-mobile/src/TxHistory/ActionsBanner.tsx

```diff
@@ -41,6 +50,22 @@ export const ActionsBanner = ({disabled = false}: {disabled: boolean}) => {
     resetForm()
   }
 
+  const handleOnSwap = () => {
+    const sellTokenInfo = tokenInfos.filter((tokenInfo) => tokenInfo.id === createOrder.amounts.sell.tokenId)[0]
+    const buyTokenInfo = tokenInfos.filter((tokenInfo) => tokenInfo.id === createOrder.amounts.buy.tokenId)[0]
```

Comment:
what to do in the case tokenInfo can't be found? product question?

Reply:
from the `TxHistory` when navigating to `Swap` it should reset the state, so you don't need to read the state here, you only need the initial. `sell` comes from balance, but `buy` comes from the api. (you can only don't have the `sell` one after pressing switch.